Cubical/Modalities/Lex.agda

```diff
@@ -0,0 +1,279 @@
+      where
+        η-inv : ◯ ((x : A) → B x) → (x : A) → B x
+        η-inv [f] x = ◯-rec (B-mod x) (λ f → f x) [f]
+
+        retr : retract η η-inv
+        retr f = funExt λ x → ◯-rec-β (B-mod x) _ f
+
+    Σ-modal : isModal A → isModalFam B → isModal (Σ A B)
+    Σ-modal A-mod B-mod = retract-is-modal idemp η-inv η retr
+      where
+        h : ◯ (Σ A B) → A
+        h = ◯-rec A-mod fst
+
+        h-β : (x : Σ A B) → h (η x) ≡ fst x
+        h-β = ◯-rec-β A-mod fst
+
+        f : (i : I) (x : Σ A B) → B (h-β x i)
+        f i x = coe1→i (λ j → B (h-β x j)) i (snd x)
+
+        η-inv : ◯ (Σ A B) → Σ A B
+        η-inv y = h y , ◯-ind (B-mod ∘ h) (f i0) y
+
+        retr : (p : Σ A B) → η-inv (η p) ≡ p
+        retr p =
+          η-inv (η p)
+            ≡⟨ pair-ext refl (◯-ind-β _ _ _) ⟩
+          h (η p) , f i0 p
+            ≡⟨ pair-ext (h-β _) (λ i → f i p) ⟩
+          p ∎
+
+
+module Σ-commute {A : Type ℓ} (B : A → Type ℓ′) where
+  open LiftFamExtra
+
+  ◯Σ = ◯ (Σ A B)
+
+  module Σ◯ where
+    Σ◯ = Σ (◯ A) (⟨◯⟩ B)
+    abstract
+      Σ◯-modal : isModal Σ◯
+      Σ◯-modal = Σ-modal idemp (⟨◯⟩-modal _)
+
+  open Σ◯
+
+  η-Σ◯ : Σ A B → Σ◯
+  η-Σ◯ (x , y) = η x , ⟨η⟩ y
+
+  module Push where
+    abstract
+      fun : ◯Σ → Σ◯
+      fun = ◯-rec Σ◯-modal η-Σ◯
+
+      compute : fun ∘ η ≡ η-Σ◯
+      compute = funExt (◯-rec-β _ _)
+
+  module Unpush where
+    abstract
+      fun : Σ◯ → ◯Σ
+      fun =
+        λ⟨,⟩ ◯-ind (λ _ → Π-modal λ _ → idemp) λ x →
+        λ/coe⟨ ⟨◯⟩-compute B x ⟩ ◯-map (x ,_)
+
+      compute : fun ∘ η-Σ◯ ≡ η
+      compute =
+        funExt λ p →
+        fun (η-Σ◯ p)
+          ≡⟨ cong-fun (◯-ind-β _ _ _) _ ⟩
+        transport refl (◯-map _ _)
+          ≡⟨ transportRefl _ ⟩
+        ◯-map _ (⟨◯⟩→◯ (⟨η⟩ _))
+          ≡⟨ cong (◯-map _) (⟨◯⟩→◯-section _) ⟩
+        ◯-map _ (η _)
+          ≡⟨ ◯-map-β _ _ ⟩
+        η p ∎
+
+
+  push-unpush-compute : Push.fun ∘ Unpush.fun ∘ η-Σ◯ ≡ η-Σ◯
+  push-unpush-compute =
+    Push.fun ∘ Unpush.fun ∘ η-Σ◯
+      ≡⟨ cong (Push.fun ∘_) Unpush.compute ⟩
+    Push.fun ∘ η
+      ≡⟨ Push.compute ⟩
+    η-Σ◯ ∎
+
+  unpush-push-compute : Unpush.fun ∘ Push.fun ∘ η ≡ η
+  unpush-push-compute =
+    Unpush.fun ∘ Push.fun ∘ η
+      ≡⟨ cong (Unpush.fun ∘_) Push.compute ⟩
+    Unpush.fun ∘ η-Σ◯
+      ≡⟨ Unpush.compute ⟩
+    η ∎
+
+  is-section : section Unpush.fun Push.fun
+  is-section = ◯-ind (λ x → ≡-modal idemp) λ x i → unpush-push-compute i x
+
+  is-retract : retract Unpush.fun Push.fun
+  is-retract =
+    λ⟨,⟩ ◯-ind (λ _ → Π-modal λ _ → ≡-modal Σ◯-modal) λ x →
+    λ/coe⟨ ⟨◯⟩-compute B x ⟩
+     ◯-ind
+      (λ _ → ≡-modal Σ◯-modal)
+      (λ y i → push-unpush-compute i (x , y))
+
+  push-sg-is-equiv : isEquiv Push.fun
+  push-sg-is-equiv = isoToIsEquiv (iso Push.fun Unpush.fun is-retract is-section)
+
+
+
+isConnected : Type ℓ → Type ℓ
+isConnected A = isContr (◯ A)
+
+
+
+module FormalDiskBundle {A : Type ℓ} where
+  𝔻 : A → Type ℓ
+  𝔻 a = Σ A (λ x → η a ≡ η x)
```

Comment:
I think the fancy Sigma notation would make this definition look nicer: https://github.com/agda/cubical/blob/master/Cubical/Core/Primitives.agda#L219